src/pybind/mgr/telemetry/tests/telemetry_upgrade_tests.md

```diff
@@ -0,0 +1,213 @@
+The `upgrade` suite is used to verify that upgrades from the (X-2) releases of
+Ceph (where X stands for the current release, i.e tentacle as of writing this
+documentation) to the current stable Ceph release (_tentacle_) can complete
+successfully without disrupting any ongoing workloads.
+
+There are two goals of this suite:
+- Test that the Ceph cluster has been successfully upgraded to the desired
+  version without the workloads (eg: basic rados api tests, import/export cli
+  tests for rbd, stress testing etc) getting disrupted
+- Test Telemetry functions as expected after the upgrade
+
+The diagram below represents the upgrade test directory from the squid release
+branch. Each release branch upgrade directory includes X-2 upgrade testing. That
+means, we can test the upgrade from 2 previous releases to the current one.
+
+```
+upgrade
+├── quincy-x
+│   ├── filestore-remove-check
+│   ├── parallel
+│   │   ├── 0-start.yaml
+│   │   ├── 1-tasks.yaml
+│   │   ├── upgrade-sequence.yaml
+│   │   └── workload
+│   └── stress-split
+|
+├── reef-x
+│   ├── parallel
+│   │   └── workload
+│   └── stress-split
+|
+├── squid-p2p
+│   ├── squid-p2p-parallel
+│   └── squid-p2p-stress-split
+|
+└── telemetry-upgrade
+    ├── quincy-x
+    └── reef-x
+
+```
+
+Based on the above example where X=squid, is possible to test the upgrades from
+Quincy (X-2) or from Reef (X-1) to Squid (X).
+
+- The `upgrade/quincy-x/parallel` and `upgrade/reef-x/parallel` sub-suite
+  installs a Quincy or Reef cluster, then upgrades the cluster to Squid (X). In
+  parallel, some workloads are run against the cluster, including telemetry
+  workunits.
+- The `upgrade/telemetry-upgrade` sub-suite is identical to
+  `upgrade/quincy-x/parallel` and `upgrade/reef-x-parallel` sub-suites above,
+  but these only test the telemetry workunits and do not run any other
+  workloads.
+
+A simple upgrade test contains these steps in order, divided into separate yaml files:
+```
+├── 0-start.yaml
+├── 1-tasks.yaml
+├── upgrade-sequence.yaml
+└── workload
+```
+
+- `0-start.yaml`: This file contains the information about the ceph cluster
+  configuration (number of osds, monitors etc) for the test
+- `1-tasks.yaml`: This file contains the information of the tasks we want to run
+  on the cluster. It is here that we install an older release, then begin
+  running the given `workload` and `upgrade-sequence` in parallel.
+- `upgrade-sequence.yaml`: This file contains the steps for upgrading the
+  cluster to the latest release
```

Comment:
```suggestion
- `upgrade-sequence.yaml`: This file contains the steps for upgrading the
  cluster to the designated release
```
In this case, Squid isn't the latest release, but it is the designated release that we are testing the upgrade sequence on.